### ISSUE TITLE
Ipad app rejected

### ISSUE THREAD

**Issue by kevinrodriguez-io** [ResearchKit/ResearchKit#734]
Hi guys,

Yesterday i had problems with the review team because my app was linking "HealthKit".

I use the researchkit only because of the Charts component, but it seems that other components do use HealthKit references.

I tried to clone and prune the project in order to be able to remove all the components which use HealthKit, but i'm having problems with the ORKAccessibilityFunctions file under Accessibility, the problem is that this file references ORKAnswerFormat.h and it does reference HealthKit.

What should i do?

Can i just explain the Review team this situation?

Regards,
Kevin

**Comment by YuanZhu-apple** [comment]
Can you remove the `ORKAnswerFormat.h` reference from `ORKAccessibilityFunctions.m`? And see if that works?

**Comment by darkndream** [comment]
Yes, i removed that reference and also removed the ORKAccessibilityFOrmatScaleSliderValue Method and the ORKAccesibilityFormatContinuousScaleSliderValue method too, also commented the ORKScaleSlider.h and the ORKScaleSliderView.h, which referenced HealthKit too,

Thanks.

I'll try to let you know if i have any success !

**Comment by YuanZhu-apple** [comment]
@darkndream Just want to confirm you only copy out the necessary header/source files from RK to your project, right?

**Comment by darkndream** [comment]
No, my project is written in swift and i installed RK via cocoapods,
I could try copying the framework directly or including RK as a subproject and only reference the Chart views but i don't know if that would make the review team happy.

Anyways i think i'm going to include it as a framework after pruning the other components, that should make get rid of any HealthKit reference and make the review team happy i think!

**Comment by darkndream** [comment]
@YuanZhu-apple We explained the review team the situation and showed them the usage of the ResearchKit and we just received their approval.

I think the problem is solved.

Thanks.

**Comment by dadixon** [comment]
What all did you remove to get Apple to pass this? I'm using just the survey and ran into Apple issues with HealthKit. I'm not using HealthKit at all. I'm trying to figure out what references you took out without breaking ResearchKit

**Comment by kevinrodriguez-io** [comment]
@dadixon back at the day I our to explain deeply this situation, which I think apple is aware of. You also need to detail the api usage for this package and why you're not using the other parts of it. Just the charts.

**Comment by dadixon** [comment]
I'll try to explain more to them. I also had to add some user permissions which I also didn't need.
